plugins/package-managers/maven/src/main/kotlin/utils/MavenDependencyHandler.kt

```diff
@@ -112,7 +114,24 @@ class MavenDependencyHandler(
                 message = "Could not get package information for dependency '" +
                     "${dependency.artifact.identifier()}': ${e.collectMessages()}"
             )
-        }.getOrNull()
+        }.getOrElse {
+            Package(
+                id = dependency.artifact.run {
+                    Identifier(
+                        type = "Maven",
+                        namespace = groupId,
+                        name = artifactId,
+                        version = version
+                    )
+                },
+                binaryArtifact = RemoteArtifact.EMPTY,
```

Comment:
I wonder why we shouldn't be able to at least get the binary artifact here... I mean, apparently software with a dependency on `opensymphony:quartz:1.6.0` *does* build, so the binary can be retrieved, and we should be able to get its coordinates here.

Maybe instead of this fallback here, can we narrow down which part of `parsePackage()` fails, and try to get other data from it on a more fine-granular basis?

Reply:
>can we narrow down which part of parsePackage() fails

The Maven call ([MavenSupport.kt#698](https://github.com/oss-review-toolkit/ort/blob/4f32b500b2d29404d2260c191f25f4aea09d5fa3/plugins/package-managers/maven/src/main/kotlin/utils/MavenSupport.kt#L698) => [MavenSupport.kt#715](https://github.com/oss-review-toolkit/ort/blob/4f32b500b2d29404d2260c191f25f4aea09d5fa3/plugins/package-managers/maven/src/main/kotlin/utils/MavenSupport.kt#L715)) raises `ProjectBuildingException`. Not sure if it's reasonable to break the contract of `parsePackage` rethrowing these. Same goes for `requestRemoteArtifact` in `MavenDependencyHandler` (given that it is a private method).

Reply:
>try to get other data from it on a more fine-granular basis

I've checked the exception and it doesn't seem like there's much to extract from it, except for the fact that the cause is `ArtifactResolutionException`. This could be used to make a conditional that tries to fetch the binary, I guess?

Another way would be to try and fetch the remote artifact before the POM, but it's probably a bad idea given how rare the case of POMless dependency is.